src/commands/database/mysql/user/create.tsx

```diff
@@ -0,0 +1,146 @@
+import { ExecRenderBaseCommand } from "../../../../lib/basecommands/ExecRenderBaseCommand.js";
+import {
+  makeProcessRenderer,
+  processFlags,
+} from "../../../../rendering/process/process_flags.js";
+import { ReactNode } from "react";
+import { Flags, Args } from "@oclif/core";
+import { assertStatus } from "@mittwald/api-client-commons";
+import { Success } from "../../../../rendering/react/components/Success.js";
+import { Value } from "../../../../rendering/react/components/Value.js";
+
+type Result = {
+  mysqlUserId: string;
+};
+
+export class Create extends ExecRenderBaseCommand<typeof Create, Result> {
+  static description = "Create a new mysql user";
+  static args = {
+    "database-id": Args.string({
+      description: "ID of the MySQL Database to create a user for",
+    }),
+  };
+  static flags = {
+    ...processFlags,
+    "database-id": Flags.string({
+      required: true,
+      description: "ID of the MySQL Database to create a user for",
+    }),
+    "access-level": Flags.string({
+      required: true,
+      description: "Access level for this mysql user",
+      options: ["readonly", "full"],
+    }),
+    description: Flags.string({
+      required: true,
+      description: "Description of the mysql user",
+    }),
+    password: Flags.string({
+      required: true,
+      description: "Password used for authentication",
+      exactlyOne: ["public-key", "password"],
+    }),
+    "access-ip-mask": Flags.string({
+      description: "IP from wich external access will be exclusively allowed",
+    }),
+    "external-access": Flags.boolean({
+      description: "Enable/Disable external access for this user.",
+    }),
+  };
+
+  protected async exec(): Promise<Result> {
+    const process = makeProcessRenderer(
+      this.flags,
+      "Creating a new mysql User",
+    );
+
+    // TODO: implement withMySQLId
+
+    let mysqlDatabaseId: string = "";
+    if (this.flags["database-id"]) {
+      mysqlDatabaseId = this.flags["database-id"];
+    } else if (this.args["database-id"]) {
+      mysqlDatabaseId = this.args["database-id"];
+    } else {
+      await process.error(
+        "Please provide a backup schedule id as flag or argument",
+      );
+    }
+
+    const {
+      "access-level": accessLevel,
+      description,
+      password,
+      "access-ip-mask": accessIpMask,
+      "external-access": externalAccess,
+    } = this.flags;
+
+    const createMysqlUserPayload: {
+      accessLevel: "full" | "readonly";
+      databaseId: string;
+      description: string;
+      password: string;
+      accessIpMask?: string | undefined;
+      externalAccess?: boolean | undefined;
```

Comment:
Can we extract these types somewhere from the API client? Or are they inlined there, which makes this duplication necessary?

Reply:
This I would be very interested in learning more about. 
The declaration of the type there helps to guide the creation of the Payload wich is then sent to the api.
these type I have copied from the data key of the (in this case) createMysqlUser function.

Two things though:
1.  since I only need the data key, the type would look something like 
```type MyQSLUserData = MittwaldAPIV2.Components.Schemas.DatabaseMySqlUser.data;```
which is syntactically not possible

```TS2713: Cannot access DatabaseMySqlUser.data because DatabaseMySqlUser is a type, but not a namespace. Did you mean to retrieve the type of the property data in DatabaseMySqlUser with 'DatabaseMySqlUser[data]'?```

If I try to retrieve the data property though, using

``

Reply:
Bleh, there actually is no named type for this method's request body.
This is hacky AF, but you can (just tried it out with `createMySQLDatabase`) do this:

```typescript
type CreateParam = Parameters<MittwaldAPIV2Client["database"]["createMysqlDatabase"]>[0]["data"];
```

Reply:
See https://github.com/mittwald/cli/pull/720, for example.

Reply:
Addendum: Using helpers like `Partial<?>`, `Omit<?>` or `Required<?>`,[^1] you can also modify the inferred types, to make properties optional or required, or omit them entirely.

[^1]: https://www.typescriptlang.org/docs/handbook/utility-types.html